src/spec.rs

```diff
@@ -0,0 +1,104 @@
+//! Helper functions defined in the Zcash Protocol Specification.
+
+use std::iter;
+
+use blake2b_simd::{Hash, Params};
+use ff::PrimeField;
+use group::Curve;
+use halo2::{
+    arithmetic::{CurveAffine, CurveExt, FieldExt},
+    pasta::pallas,
+};
+
+use crate::{constants::L_ORCHARD_SCALAR, primitives::sinsemilla};
+
+const PRF_EXPAND_PERSONALIZATION: &[u8; 16] = b"Zcash_ExpandSeed";
+
+/// $\mathsf{ToBase}^\mathsf{Orchard}(x) := LEOS2IP_{\ell_\mathsf{PRFexpand}}(x) (mod q_P)$
+///
+/// Defined in [Zcash Protocol Spec § 4.2.3: Orchard Key Components][orchardkeycomponents].
+///
+/// [orchardkeycomponents]: https://zips.z.cash/protocol/nu5.pdf#orchardkeycomponents
+pub(crate) fn to_base(hash: Hash) -> pallas::Base {
+    pallas::Base::from_bytes_wide(hash.as_array())
+}
+
+/// $\mathsf{ToScalar}^\mathsf{Orchard}(x) := LEOS2IP_{\ell_\mathsf{PRFexpand}}(x) (mod r_P)$
+///
+/// Defined in [Zcash Protocol Spec § 4.2.3: Orchard Key Components][orchardkeycomponents].
+///
+/// [orchardkeycomponents]: https://zips.z.cash/protocol/nu5.pdf#orchardkeycomponents
+pub(crate) fn to_scalar(hash: Hash) -> pallas::Scalar {
+    pallas::Scalar::from_bytes_wide(hash.as_array())
+}
+
+/// Defined in [Zcash Protocol Spec § 4.2.3: Orchard Key Components][orchardkeycomponents].
+///
+/// [orchardkeycomponents]: https://zips.z.cash/protocol/nu5.pdf#orchardkeycomponents
+pub(crate) fn commit_ivk(
+    ak: &pallas::Base,
```

Comment:
```suggestion
    ak: &pallas::Base,
```
In the spec this will be a point. (It's currently wrong in the draft.)